src/stream/src/executor/rearranged_chain.rs

```diff
@@ -135,10 +141,11 @@ impl RearrangedChainExecutor {
     async fn execute_inner(mut self) {
         // 0. Project the upstream with `upstream_indices`.
         let upstream_indices = self.upstream_indices.clone();
-        let mut upstream = self
-            .upstream
-            .execute()
-            .map(move |result| result.map(|msg| mapping(&upstream_indices, msg)));
```

Comment:
We should be able to use `.map(..).flatten()` here.

Reply:
Could you please elaborate a bit more？

Reply:
```rust
        let mut upstream = self
            .upstream
            .execute()
            .map(move |result| result.map(|msg| mapping(&upstream_indices, msg)))
            .flatten();
```

Reply:
But `mapping` returns an Option, which seems unable to be flattened.